Fix charging status enum translation

jtchitty opened the pull request:
Handle the previously unhandled 'Not charging' status from the driver.
This is an error condition that should translate to 'CHARGEERROR',
meaning that there is an error condition preventing charging.

Translate failure to read the charging status into 'HEALTHUNDEFINED'
instead of 'CHARGEERROR' because being unable to read the status
does not necessarily mean that charging isn't working.

Commit:
Fix charging status enum translation

Handle the previously unhandled 'Not charging' status from the driver.
This is an error condition that should translate to 'CHARGEERROR',
meaning that there is an error condition preventing charging.

Translate failure to read the charging status into 'HEALTHUNDEFINED'
instead of 'CHARGEERROR' because being unable to read the status
does not necessarily mean that charging isn't working.

dpfrey reviewed (commented):
Seems reasonable, but I think Ashish should give feedback too.

ashsyal commented:
I dont believe NOT_CHARGING should be reported as chargerror. If no charger is connected, then battery is not being charged. We cant report that as error

ashsyal commented:
MA_BATTERY_HEALTHUNDEFINED should be  MA_BATTERY_UNKNOWN. This is not a health status.
Also, power supply  enum in powersupply.h, shows the following enum definition:
enum {
	POWER_SUPPLY_STATUS_UNKNOWN = 0,
	POWER_SUPPLY_STATUS_CHARGING,
	POWER_SUPPLY_STATUS_DISCHARGING,
	POWER_SUPPLY_STATUS_NOT_CHARGING,
	POWER_SUPPLY_STATUS_FULL,
};

jtchitty commented:
> I dont believe NOT_CHARGING should be reported as chargerror. If no charger is connected, then battery is not being charged. We cant report that as error

In the Data Hub, a disconnected battery shows up as `"health":"disconnected"`.
No error is reported in that case.

In the API, you'd get CHARGEERROR if you tried to fetch the charging status for a disconnected battery, but that makes sense to me.  There's no battery, so why are you asking for the charging status?  Check the health status first and it will tell you there's no battery.

jtchitty commented:
> MA_BATTERY_HEALTHUNDEFINED should be MA_BATTERY_UNKNOWN. This is not a health status.

Looks like this bug was copied over from the Red version.  I'll fix it in both places.

Commit:
Clean up naming of charging and health enums

Add NOT_CHARGING to the ChargingStatus enum.
Change CHARGEERROR to CHARGING_ERROR.
Change UNKNOWN to CHARGING_UNKNOWN.
Change HEALTHERROR to HEALTH_ERROR.
Change HEALTHUNDEFINED to HEALTH_UNKNOWN.
Clarified comment for CHARGING_ERROR.
Fix bug on mangOH Red where charging error code was returned for
health status error.
